client.go

```diff
@@ -603,6 +615,12 @@ func (c *Client) execute(req *Request) (*Response, error) {
 
 	req.Time = time.Now()
 	c.httpClient.Transport = c.transport
+
+	if req.RawRequest.URL != nil && req.RawRequest.URL.Scheme == "" {
+		req.RawRequest.URL.Scheme = c.scheme
+		req.RawRequest.URL.Host = req.URL
```

Comment:
@jeevatkm Not for my use case.

The requested URL won't contain any `Host`, that's why I had to set up the `Scheme` like that since the transport is given by me.

This allows go-resty to be used with unix sockets.